apps/busybox/tests/pvc_creation/main.yml

```diff
@@ -0,0 +1,91 @@
+---
+#### Required changes #########
+
+   - name: Get the application label values from env
+     set_fact:
+         app_lkey: "{{ app_label.split('=')[0] }}"
+         app_lvalue: "{{ app_label.split('=')[1] }}"
+
+   - name: Replace the application label placeholder in spec
+     replace:
+        path: "{{ application_deployment }}"
+        regexp: "lkey: lvalue"
+        replace: "{{ app_lkey }}: {{ app_lvalue }}{{i}}"
+ 
+   - name: Change the claim name in deployment.
+     replace:
+       dest: "{{application_deployment}}"
+       regexp: "{{app_pvc}}"
+       replace: "{{app_pvc}}-{{i}}"
+   
+   - name: Change the app name in deployment.
+     replace:
+       dest: "{{application_deployment}}"
+       regexp: "{{app_name}}"
+       replace: "{{app_name}}-{{i}}"
+   
+   - block:
+       - name: Create namespace for deployment.
+         shell: kubectl create ns {{ namespace }}-{{i}}
+         args:
+           executable: /bin/bash
+         register: app_ns
+         failed_when: "'created' not in app_ns.stdout"
+
+       ## Deploying the application (statefulset/deployment) and its related services.
+       - include_tasks: /common/utils/deploy_single_app.yml
+         vars:
+           app_ns: "{{ namespace }}-{{ i }}"
+           check_app_pod: 'no'
+           delay: 10
+           retries: 20
+
+       - include_tasks: /common/utils/check_deployment_status.yml  
+         vars:
+           app_ns: "{{namespace}}-{{i}}" 
+           app_label: "{{ app_lkey }}={{ app_lvalue }}{{i}}"
+
+     when: lookup('env', 'ACTION') == 'create'
+
+       ## Cleanup tasks for application  deployments ##  
+   - block:
+   
+       - name: Delete application and its deployments.
+         shell: kubectl delete -f {{application_deployment}} -n {{namespace}}-{{i}}
+         args:
+           executable: /bin/bash
+
```

Comment:
Add validation steps for pvc and pod deletion.

Reply:
Also rename the main.yaml to pvc-provision.yaml

Reply:
sure @dargasudarshan